hip_data_tools/etl/athena_to_adwords.py

```diff
@@ -0,0 +1,162 @@
+"""
+handle ETL of data from Athena to Cassandra
+"""
+import logging as log
+from datetime import datetime
+from typing import List
+
+from attr import dataclass
+from cassandra.cqlengine import ValidationError
+from pandas import DataFrame
+
+from hip_data_tools.apache.cassandra import CassandraConnectionManager, CassandraConnectionSettings
+from hip_data_tools.etl.athena_to_dataframe import AthenaToDataFrame, AthenaToDataFrameSettings
+from hip_data_tools.etl.common import EtlSinkRecordStateManager, sync_etl_state_table, EtlStates
+from hip_data_tools.google.adwords import AdWordsOfflineConversionUtil, \
+    GoogleAdWordsConnectionManager, GoogleAdWordsConnectionSettings
+
+
+@dataclass
+class AthenaToAdWordsOfflineConversionSettings(AthenaToDataFrameSettings):
+    """S3 to Cassandra ETL settings"""
+    transformation_column_mapping: dict
+    etl_identifier: str
+    etl_state_manager_keyspace: str
+    etl_state_manager_connection: CassandraConnectionSettings
+    destination_batch_size: int
+    destination_connection_settings: GoogleAdWordsConnectionSettings
+
+
+def _get_record_signature(record: dict):
+    return f"{record['googleClickId']}||||{record['conversionName']}"
```

Comment:
what happens when the same gclld and conversion metric comes across on two different timestamps. My understanding is that this would be covered as different conversions. Should the conversionTime be part of the record_signature. Can you add a testcase for a second conversion coming though on a different date? What is expected to happen?

Reply:
No, only the first record is uploaded others are rejected

> When a conversion for the same google click ID, conversion name and conversion time is uploaded multiple times, the first one results in a conversion being recorded. The duplicates are ignored and reported as successes, to indicate that a conversion for this combination has been recorded. 

As noted in https://developers.google.com/adwords/api/docs/reference/v201809/OfflineConversionFeedService.OfflineConversionFeed
under heading conversionTime

Reply:
fixd